### DISCUSSION TITLE
Swift & C++ interop

### DISCUSSION THREAD

**Discussion by mfazekas** [reactwg/react-native-new-architecture#133] [Q&A]
In order to implement Fabric component in swift. I've tried to get swift C++ interop working with 

```js
  s.pod_target_xcconfig = {
    'OTHER_SWIFT_FLAGS' => '-enable-experimental-cxx-interop'
  }
```

But I was getting error:
```
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/fabric_test_lib-umbrella.h"

...

ios/Pods/Headers/Public/React-Fabric/react/renderer/core/EventListener.h:38:8: error: no type named 'shared_mutex' in namespace 'std'; did you mean 'timed_mutex'?
  std::shared_mutex mutex_;
  
  /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk/usr/include/c++/v1/mutex:236:24: note: 'timed_mutex' declared here
class _LIBCPP_TYPE_VIS timed_mutex
```

Will try again with new beta Xcode 15 beta as swift 5.9 seems to have a much improved C++ interop. 

https://developer.apple.com/videos/play/wwdc2023/10172
https://forums.swift.org/t/c-interoperability-in-swift-5-9/65369
https://github.com/apple/swift-evolution/blob/main/visions/using-c%2B%2B-from-swift.md

**Comment by TheRogue76** [comment]
Mostly for anyone interested, here are some samples apple put out for how it works:
[https://github.com/apple/swift-cmake-examples](https://github.com/apple/swift-cmake-examples)
